### PR TITLE
How Mattermost stores Active Directory ObjectGUID.

### PR DESCRIPTION
**Pull request by fakela**

### CODE REVIEW

**Comment by fakela** [comment]
``powershell
> get-adgroup -identity developers

DistinguishedName : CN=developers,CN=Users,DC=example,DC=com
GroupCategory     : Security
GroupScope        : Universal
Name              : developers
ObjectClass       : group
ObjectGUID        : 1effbc1e-4aa3-4f9d-a508-d4d81f1f5873
SamAccountName    : developers
SID               : S-1-5-21-3389680124-1638897943-855603155-1112
You can convert this GUID format in powershell like this:

> $guid = [System.Guid]"{1effbc1e-4aa3-4f9d-a508-d4d81f1f5873}"
> $guid.ToByteArray()
30
188
255
30
163
74
157
79
165
8
212
216
31
31
88
115


I would love to know how this part of the information should be entered

File: source/install/troubleshooting.rst [hunk 1]
```diff
@@ -311,6 +311,18 @@ This error message applies to various validation issues. The log message provide
 
 - ``too soon`` or ``too late``: Assertion ``NotOnOrAfter`` or ``NotBefore`` attribute outside current time.
 
+``Why does the objectGUID of a user in Mattermost differ from what we're seeing in ADFS``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The Active Directory Object-Guid attribute LDAP display name ``objectGUID`` is a 16 byte array which can be displayed in different ways:
```
Comment (justinegeffen) [inline source/install/troubleshooting.rst:317]:
```suggestion
The Active Directory ObjectGUID attribute LDAP display name ``objectGUID`` is a 16-byte array which can be displayed in different ways:
```

File: source/install/troubleshooting.rst [hunk 2]
```diff
@@ -311,6 +311,18 @@ This error message applies to various validation issues. The log message provide
...
+
+- The ``ldapsearch`` linux command displays it as base 64: ``Hrz/HqNKnU+lCNTYHx9Ycw==``. This is also the format used in LDIF files.
```
Comment (justinegeffen) [inline source/install/troubleshooting.rst:319]:
```suggestion
- The ``ldapsearch`` Linux command displays it as base64: ``Hrz/HqNKnU+lCNTYHx9Ycw==``. This is also the format used in LDIF files.
```

File: source/install/troubleshooting.rst [hunk 3]
```diff
@@ -311,6 +311,18 @@ This error message applies to various validation issues. The log message provide
...
+
+- The `LDAP Golang package we use <https://github.com/go-ldap/ldap>`__ shows the value as hexidecimal (base 16) array with each byte separated by a backslash: ``\1e\bc\ff\1e\a3\4a\9d\4f\a5\08\d4\d8\1f\1f\58\73``
```
Comment (justinegeffen) [inline source/install/troubleshooting.rst:321]:
```suggestion
- The `LDAP Golang package we use <https://github.com/go-ldap/ldap>`__ shows the value as hexadecimal (base 16) array with each byte separated by a backslash: ``\1e\bc\ff\1e\a3\4a\9d\4f\a5\08\d4\d8\1f\1f\58\73``.
```

File: source/install/troubleshooting.rst [hunk 4]
```diff
@@ -311,6 +311,18 @@ This error message applies to various validation issues. The log message provide
...
+
+You can remove the backslashes: ``1ebcff1ea34a9d4fa508d4d81f1f5873`` and parse it with `Golang <https://play.golang.org/p/9b8iDPuz0Nm>`__ as snippets which prints the base 10 representation of each value: ``30 188 255 30 163 74 157 79 165 8 212 216 31 31 88 115``
```
Comment (justinegeffen) [inline source/install/troubleshooting.rst:323]:
```suggestion
You can remove the backslashes: ``1ebcff1ea34a9d4fa508d4d81f1f5873`` and parse it with `Golang <https://play.golang.org/p/9b8iDPuz0Nm>`__ as snippets which prints the base 10 representation of each value: ``30 188 255 30 163 74 157 79 165 8 212 216 31 31 88 115``.
```

File: source/install/troubleshooting.rst [hunk 5]
```diff
@@ -311,6 +311,18 @@ This error message applies to various validation issues. The log message provide
...
+
+- Windows powershell displays the value like this: ``1effbc1e-4aa3-4f9d-a508-d4d81f1f5873``
```
Comment (justinegeffen) [inline source/install/troubleshooting.rst:325]:
```suggestion
- Windows PowerShell displays the value like this: ``1effbc1e-4aa3-4f9d-a508-d4d81f1f5873``.
```

**Comment by justinegeffen** [comment]
> ``powershell
> 
> > get-adgroup -identity developers
> 
> DistinguishedName : CN=developers,CN=Users,DC=example,DC=com
> GroupCategory : Security
> GroupScope : Universal
> Name : developers
> ObjectClass : group
> ObjectGUID : 1effbc1e-4aa3-4f9d-a508-d4d81f1f5873
> SamAccountName : developers
> SID : S-1-5-21-3389680124-1638897943-855603155-1112
> You can convert this GUID format in powershell like this:
> 
> > $guid = [System.Guid]"{1effbc1e-4aa3-4f9d-a508-d4d81f1f5873}"
> > $guid.ToByteArray()
> > 30
> > 188
> > 255
> > 30
> > 163
> > 74
> > 157
> > 79
> > 165
> > 8
> > 212
> > 216
> > 31
> > 31
> > 88
> > 115
> 
> I would love to know how this part of the information should be entered

Thanks, @fakela. I've added some comments to the new content. Regarding the formatting you've mentioned above, is that how you'd like it to render? I think in this case we'd use .. code-block:: sh and then it should render correctly.

**Comment by fakela** [comment]
> > ``powershell
> > > get-adgroup -identity developers
> > 
> > 
> > DistinguishedName : CN=developers,CN=Users,DC=example,DC=com
> > GroupCategory : Security
> > GroupScope : Universal
> > Name : developers
> > ObjectClass : group
> > ObjectGUID : 1effbc1e-4aa3-4f9d-a508-d4d81f1f5873
> > SamAccountName : developers
> > SID : S-1-5-21-3389680124-1638897943-855603155-1112
> > You can convert this GUID format in powershell like this:
> > > $guid = [System.Guid]"{1effbc1e-4aa3-4f9d-a508-d4d81f1f5873}"
> > > $guid.ToByteArray()
> > > 30
> > > 188
> > > 255
> > > 30
> > > 163
> > > 74
> > > 157
> > > 79
> > > 165
> > > 8
> > > 212
> > > 216
> > > 31
> > > 31
> > > 88
> > > 115
> > 
> > 
> > I would love to know how this part of the information should be entered
> 
> Thanks, @fakela. I've added some comments to the new content. Regarding the formatting you've mentioned above, is that how you'd like it to render? I think in this case we'd use .. code-block:: sh and then it should render correctly.

Thanks a lot, I was a bit confused on how the information should be entered.

**Pull request merged**